components/PlaylistVisual.tsx

```diff
@@ -1,21 +1,30 @@
 import React, {useRef, useEffect} from 'react';
 import * as THREE from 'three';
 import styles from '../styles/lobby.module.scss';
-import {sampleShader} from '../utils';
+import {getElementSizeById, sampleShader} from '../utils';
 
 interface PlaylistVisualProps {
   title: string;
   subtitle: string;
-  width?: number;
-  height?: number;
 }
 
-function PlaylistVisual({title, subtitle, width, height}: PlaylistVisualProps): JSX.Element {
+function PlaylistVisual({
+  title,
+  subtitle,
+}: PlaylistVisualProps): JSX.Element {
   const ref = useRef(null);
 
+  const getVisualSize = () => {
+    return getElementSizeById(styles.visual, {
+      width: parseInt(styles.defaultVisualWidth),
+      height: parseInt(styles.defaultVisualHeight),
+    });
+  };
+
   useEffect(() => {
-    const w = width ?? window.innerWidth;
-    const h = height ?? window.innerHeight;
+    const size = getVisualSize();
+    const w = size != null ? size.width : window.innerWidth;
+    const h = size != null ? size.height : window.innerHeight;
```

Comment:
nit, i think this is cleaner

```suggestion
    const w = size?.width ?? window.innerWidth;
    const h = size?.height ?? window.innerHeight;
```